### PR TITLE
10x speed increase for resolution.

### PR DESCRIPTION
**Pull request by EricBishton**
This batch of changes introduces a number of incremental speed increases that together make resolution (thus, annotation, etc.) 10x faster.  The main impetus was that changes in large files (or importing large files) force the CPU to run very high for minutes on end.  This is very apparent when used with the Nape physics libraries.  (See issue #851).

### CODE REVIEW

**Comment by EricBishton** [comment]
Tests are failing.  Glad we have them!  @mayakwd Don't bother with the review until I've got them figured out.

**Comment by EricBishton** [comment]
@mayakwd Everything is working.  Please review.

File: src/common/com/intellij/plugins/haxe/lang/util/HaxeAstUtil.java [hunk 1]
```diff
@@ -128,7 +128,13 @@ public static boolean isInteger(@Nullable String s) {
   public static boolean isNumber(@Nullable ASTNode t) {
     if (t != null) {
       String text = t.getText();
-      return isInteger(text) || isFloat(text);
+      // Optimization: If it doesn't start with '+', '-', or digit, it's not a number (ignoring 'Nan' and Infinity).
+      // Using the Float class to detect numbers is fairly expensive.  Before this optimization, about 2/3 of the
+      // time we spent evaluating a conditional expression was spent in isFloat().
+      char firstChar = text.charAt(0);
+      if (Character.isDigit(firstChar) || '+' == firstChar || '-' == firstChar) {
```
Comment (mayakwd) [inline src/common/com/intellij/plugins/haxe/lang/util/HaxeAstUtil.java]: Float could be written as `.025` that equals `0.025`

Comment (EricBishton) [reply src/common/com/intellij/plugins/haxe/lang/util/HaxeAstUtil.java]: Ah, yes.  Thanks!

**Pull request merged**